pkg/apis/garden/validation/validation.go

```diff
@@ -1760,6 +1766,13 @@ func ValidateShootSpecUpdate(newSpec, oldSpec *garden.ShootSpec, deletionTimesta
 		allErrs = append(allErrs, apivalidation.ValidateImmutableField(newSpec.Cloud.Azure.Networks.VNet, oldSpec.Cloud.Azure.Networks.VNet, azurePath.Child("networks", "vnet"))...)
 		allErrs = append(allErrs, apivalidation.ValidateImmutableField(newSpec.Cloud.Azure.Networks.Workers, oldSpec.Cloud.Azure.Networks.Workers, azurePath.Child("networks", "workers"))...)
 		allErrs = append(allErrs, apivalidation.ValidateImmutableField(newSpec.Cloud.Azure.ResourceGroup, oldSpec.Cloud.Azure.ResourceGroup, azurePath.Child("resourceGroup"))...)
+
+		// Don't allow to move from a zoned cluster to a non zoned cluster or the other way.
+		if len(oldSpec.Cloud.Azure.Zones) > 0 && len(newSpec.Cloud.Azure.Zones) == 0 {
```

Comment:
Can you please write tests for the validations you implemented here?

Reply:
yep, done.